app/lib/features/pins/pages/create_pin_page.dart

```diff
@@ -0,0 +1,353 @@
+      children: <Widget>[
+        const Padding(
+          padding: EdgeInsets.only(bottom: 5),
+          child: Text('Link'),
+        ),
+        InputTextField(
+          hintText: 'https://',
+          key: CreatePinPage.urlFieldKey,
+          textInputType: TextInputType.url,
+          controller: _linkController,
+          validator: (value) =>
+              hasLinkOrText() ? null : 'Text or URL must be given',
+        ),
+      ],
+    );
+  }
+
+  Widget _buildDescriptionField() {
+    final labFeature = ref.watch(featuresProvider);
+    bool isActive(f) => labFeature.isActive(f);
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        const Padding(
+          padding: EdgeInsets.only(bottom: 5),
+          child: Text('Description'),
+        ),
+        Visibility(
+          visible: isActive(LabsFeature.pinsEditor),
+          replacement: SizedBox(
+            height: 200,
+            child: MdEditorWithPreview(
+              key: CreatePinPage.contentFieldKey,
+              validator: (value) =>
+                  hasLinkOrText() ? null : 'Text or URL must be given',
+              controller: _textController,
+            ),
+          ),
+          child: Container(
+            height: 200,
+            margin: const EdgeInsets.symmetric(vertical: 8),
+            decoration: BoxDecoration(
+              color: Theme.of(context).colorScheme.primaryContainer,
+              borderRadius: BorderRadius.circular(12),
+            ),
+            constraints:
+                BoxConstraints(maxHeight: MediaQuery.of(context).size.height),
+            child: HtmlEditor(
+              editable: true,
+              autoFocus: false,
+              shrinkWrap: true,
+              editorState: EditorState.blank(),
+              footer: const SizedBox(),
+              onChanged: (body, html) {
+                if (html != null) {
+                  _textController.text = html;
+                }
+                _textController.text = body;
+              },
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  bool hasLinkOrText() {
+    return _linkController.text.trim().isNotEmpty ||
+        _textController.text.trim().isNotEmpty;
+  }
+
+  void _handleCreatePin() async {
+    EasyLoading.show(status: 'Creating pin...');
+    try {
+      final client = ref.read(alwaysClientProvider);
+      final spaceId = ref.read(selectedSpaceIdProvider);
+      final space = await ref.read(spaceProvider(spaceId!).future);
+      final pinDraft = space.pinDraft();
+      final title = _titleController.text;
+      final text = _textController.text;
+      final url = _linkController.text;
+
+      if (title.trim().isNotEmpty) {
+        pinDraft.title(title);
+      }
+
+      if (text.isNotEmpty) {
+        if (isHTML(text)) {
+          pinDraft.contentHtml(text, text);
+        } else {
+          pinDraft.contentMarkdown(text);
+        }
+      }
```

Comment:
Needs to be changed accordingly:

```suggestion

      if (_htmlTextController.text.isNotEmpty) {
          pinDraft.contentHtml(_textController.text, _htmlTextController.text);
        } else if (_textController.text.isNotEmpty) {
          pinDraft.contentPlain(_textController.text);
        }
      }
```

Reply:
Did the improvement here, check it :)
https://github.com/acterglobal/a3/pull/1457/commits/176a63dd54aef1d518e7581113dd6f863d9c9cac